analyzers/diagnostics/diagnostics.go

```diff
@@ -0,0 +1,149 @@
+	diagnostics := []Diagnostic{}
+
+	lines := strings.Split(string(fileContent), "\n")
+
+	// Iterate over each error and check line-by-line.
+	for _, err := range errors {
+		for lineNum, line := range lines {
+			result := checkField(line, err.Field)
+			if result {
+				// If the line contains the field name, and if it doesn't have a comment prefix, then we can proceed to diagnostic generation.
+				// if strings.Contains(line, err.Field) && !strings.HasPrefix(line, "#") {
+				// Prepare code frame for the current line.
+				codeFrame := prepareCodeFrame(lineNum, lines)
+
+				// Generate a diagnostic.
+				diag := Diagnostic{
+					Line:         lineNum,
+					Codeframe:    codeFrame,
+					ErrorMessage: err.Message,
+				}
+
+				diagnostics = append(diagnostics, diag)
+			}
+		}
+	}
+
+	return diagnostics
+}
+
+// prepareCodeFrame prepares a code frame using the file content. The code frame is meant to be displayed on the console while reporting diagnostics.
+// NOTE: lineNum always starts from 0.
+func prepareCodeFrame(lineNum int, lines []string) string {
+	frame := ""
+
+	if lineNum <= 1 {
+		// Case 1: When the line is near the top of the file.
+		// Generate a frame with the current and next line only.
+		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum+1, lines[lineNum]))
+		frame += fmt.Sprintf("  %d | %s\n", lineNum+2, lines[lineNum+1])
+
+	} else if lineNum >= (len(lines) - 1) {
+		// Case 2: When the line is near the bottom of the file.
+		// Generate a frame with the current line only.
+		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum, lines[lineNum-1]))
+	} else {
+		// Case 3: When the line is in between the top and the bottom.
+		// Generate a frame with the the previous, current and the next line.
+		frame += fmt.Sprintf("  %d | %s\n", lineNum, lines[lineNum-1])
+		frame += aec.LightRedF.Apply(fmt.Sprintf("> %d | %s\n", lineNum+1, lines[lineNum]))
+		frame += fmt.Sprintf("  %d | %s\n", lineNum+2, lines[lineNum+1])
+	}
+
+	return frame
+}
+
+// checkField checks if the line contains the field.
+func checkField(line, field string) bool {
+	// We use fieldRegexp for checking if the current line is a field or not.
+	// If it's not a field, for example, comments for explanation, etc., then checkField should return false.
+	exp := regexp.MustCompile(fieldRegexp)
+
+	// Get index for the "field" named group.
+	matches := exp.FindStringSubmatch(line)
+	idx := exp.SubexpIndex("field")
+
+	// Capture match.
+	match := ""
+	if len(matches) != 0 && idx != -1 {
+		match = matches[idx]
+	}
+	match = strings.TrimSpace(match)
+
+	if strings.HasPrefix(match, "#") && strings.Contains(match, field) {
+		// Return true if the current line is a comment, and contains the field.
+		// Useful for scenarios where the user tends to comment out fields.
+		return true
+	} else if strings.Contains(match, field) {
+		// If the current is not a comment, but if it contains the field, return true.
+		return true
```

Comment:
These conditions can be clubbed using `||` operator.

Reply:
Added as suggestion.

Reply:
Fixed in [1fb50b8](https://github.com/deepsourcelabs/cli/pull/169/commits/1fb50b8cbe89537548b46b996d7b69fd84f41c95). Resolving.

Reply:
I tried it out, but it doesn't work in some cases. The tests fail in https://github.com/deepsourcelabs/cli/pull/169/commits/1fb50b8cbe89537548b46b996d7b69fd84f41c95.